# 점수와 선택 기준 실험

다섯 학생의 평균과 기준 이상 점수를 비교합니다. 자체 구성한 100점 만점 점수이며 실제 학생 자료가 아닙니다. Python 표준 기능만 사용합니다.

## 입력

`scores`와 `threshold`를 수정한 뒤 아래 계산·출력 셀까지 다시 실행합니다.

In [1]:
scores = [82, 75, 45, 90, 61]
threshold = 60

In [2]:
if not scores:
    raise ValueError("scores must not be empty")

mean_score = sum(scores) / len(scores)
selected = [score for score in scores if score >= threshold]

In [3]:
print("count:", len(scores))
print("mean:", mean_score)
print("threshold:", threshold)
print("selected:", selected)

count: 5
mean: 70.6
threshold: 60
selected: [82, 75, 90, 61]


## 해석

기본 입력의 평균은 70.6점입니다. 기준 60 이상은 82, 75, 90, 61점이며, 45점은 제외됩니다. 기준은 선택 결과에만 적용되므로 전체 평균은 바뀌지 않습니다.

## 조건 변경

1. 기준을 80으로 바꾸면 선택 결과는 `[82, 90]`입니다.
2. 마지막 점수를 100으로 바꾸면 평균은 78.4입니다.
3. 빈 목록은 `ValueError`를 발생시킵니다. 오류 뒤 남은 이전 출력을 새 결과로 읽지 않습니다.

해석도 수정한 뒤 커널을 재시작하고 위에서 아래로 실행합니다. 결과와 해석을 확인한 후 저장합니다.